extensions/barrage/src/main/java/io/deephaven/extensions/barrage/util/PythonTableDataService.java

```diff
@@ -204,60 +238,62 @@ private void processNewPartition(
 
             final PrimitiveIterator.OfLong bufferInfoIter = ArrowToTableConverter.extractBufferInfo(batch);
 
-            // populate the partition values
-            for (int ci = 0; ci < schema.fieldsLength(); ++ci) {
-                try (final WritableChunk<Values> columnValues = readers.get(ci).readChunk(
+            // extract partitioning values and box them to be used as Comparable in the map
+            for (int ci = 0; ci < partitioningValuesSchema.fieldsLength(); ++ci) {
+                try (final WritableChunk<Values> columnValues = readers[ci].readChunk(
                         fieldNodeIter, bufferInfoIter, recordBatchMessageInfo.inputStream, null, 0, 0)) {
 
                     if (columnValues.size() != 1) {
                         throw new IllegalArgumentException("Expected Single Row: found " + columnValues.size());
                     }
 
-                    // partition values are always boxed to make partition value comparisons easier
-                    try (final ChunkBoxer.BoxerKernel boxer =
-                            ChunkBoxer.getBoxer(columnValues.getChunkType(), columnValues.size())) {
-                        // noinspection unchecked
-                        final ObjectChunk<Comparable<?>, ? extends Values> boxedValues =
-                                (ObjectChunk<Comparable<?>, ? extends Values>) boxer.box(columnValues);
-                        partitionValues.put(schema.fields(ci).name(), boxedValues.get(0));
-                    }
+                    partitioningValuesUnordered.put(
+                            partitioningValuesSchema.fields(ci).name(), ChunkBoxer.boxedGet(columnValues, 0));
                 } catch (final IOException unexpected) {
                     throw new UncheckedDeephavenException(unexpected);
                 }
             }
 
-            listener.accept(new TableLocationKeyImpl(tableLocationKey.locationKey, partitionValues));
+            // partitioning values must be in the same order as the partitioning keys
```

Comment:
You suggested a simpler strategy, by initializing a `LinkedHashMap` in the original order.